packages/shared-lib/src/lib/__tests__/JobQueueWithClasses.spec.ts

```diff
@@ -0,0 +1,343 @@
+import { JobQueueWithClasses } from '../JobQueueWithClasses'
+import { sleep } from '../lib'
+
+describe('JobQueueWithClasses', () => {
+	test('queue jobs synchronously', async () => {
+		const queue = new JobQueueWithClasses()
+
+		const results: string[] = []
+
+		queue.add(async () => {
+			results.push('start_0')
+			await sleep(10)
+			results.push('end_0')
+		})
```

Comment:
I guess it would be good to `.catch()` all of these, so that the linting doesn't complain.